### PR TITLE
Add multi-arch image builds supports

### PR DESCRIPTION
**Pull request by morlay**
** Describe the change **

_Please describe what the code change is about_

* Merge `Dockerfile-ubi{7|8}-minimal` to single `Dockerfile` (and make it support `docker build` & `docker buildx build` both)
* added `DORP=dockerx` to build mulit-arch image by [buildx](https://docs.docker.com/buildx/working-with-buildx/)

```bash
TARGET_ARCH="amd64 arm64 ppc64le" DORP=dockerx make build-linux container-build-kiali container-push-kiali-quay
```

example: https://hub.docker.com/r/morlay/kiali/tags

** Issue reference **

#1091

### CODE REVIEW

File: make/Makefile.container.mk
Comment (morlay) [inline make/Makefile.container.mk:36]: need to pass `TARGETARCH` for `docker build`

File: make/Makefile.container.mk
Comment (morlay) [inline make/Makefile.container.mk:14]: build multi-arch binaries.

File: deploy/docker/Dockerfile
Comment (morlay) [inline deploy/docker/Dockerfile:4]: For switch base image.

default value here to support `docker buildx build` && `docker build` both.

`docker build` could not pass arg `TARGETARCH` in `FROM`

Comment (morlay) [reply deploy/docker/Dockerfile:4]: `s390x` or `ppc64le` support could just add.
```
FROM registry.access.redhat.com/ubi8-minimal AS base-s390x
FROM registry.access.redhat.com/ubi8-minimal AS base-ppc64le
```

File: make/Makefile.container.mk
Comment (jmazzitelli) [inline make/Makefile.container.mk:14]: is this going to run "make build" multiple times? Is there any way we can default to NOT do a multi-arch build by default? Devs doing this aren't going to want to run multi-arch builds in their code-test cycles. Multi-arch builds should only be "turned on" during CI or release builds.

Also, never use `make` directly in a Makefile. Use `$(MAKE)`.

Finally, as a general "something doesn't look right" - why is "make build" being run here inside `.prepare-kiali-image-files` anyway? The targets in `Makefile.container.mk` should assume the builds are done - builds are done in `make/Makefile.build.mk` I believe. I haven't looked closely at this yet, but I suspect there is a better way to implement this to be more consistent with the way the makefiles are designed to work.

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
I haven't looked closely yet, but I think this can be implemented better. I look closely later, but for now, I'll mark this as "request changes" so we don't merge this yet until we get more eyes and reviews on this PR.

File: make/Makefile.container.mk
Comment (morlay) [reply make/Makefile.container.mk:14]: For multi arch images build. We need multi binaries for each arch `kiali-arm64` and `kiali-arm64`. 
so have build for multi-times with the arch suffix.

we could set `TARGET_ARCH ?= amd64` and `TARGET_ARCH="amd64 arm64"` in ci

We could add `make build-linux` in `make/Makefile.build.mk`
```
build-linux:
	for arch in ${TARGET_ARCH}; do \
      	    GOOS=linux GOARCH=$${arch} $(MAKE) build; \
            mv ${GOPATH}/bin/kiali ${GOPATH}/bin/kiali-$${arch}; \
        done
```

then final script in ci could be.
```
TARGET_ARCH="amd64 arm64" DORP=dockerx make build-linux container-build-kiali container-push-kiali-quay 
```

Comment (morlay) [reply make/Makefile.container.mk]: @jpkrohling Updated changes i describe.

Comment (jpkrohling) [reply make/Makefile.container.mk]: I think you meant @jmazzitelli :-)

Comment (morlay) [reply make/Makefile.container.mk]: @jpkrohling sorry. 

btw. i think Makefiles in kiali project are good examples to switch different docker image build tools. 

if you like this way too. i could update it for jaeger

**Comment by jmazzitelli** [comment]
I did a quick look again at the latest changes and these look better. I'll run some  tests as part of a further review.

File: deploy/docker/Dockerfile
Comment (jmazzitelli) [inline deploy/docker/Dockerfile:5]: I noticed the var name here is "TARGETARCH" but the Makefile uses "TARGET_ARCH" (notice the underscore in the Makefile var). I personally would prefer them to all use the same thing - so for consistency, make them all one or the other (I would vote on making them all "TARGET_ARCH" unless there is specific reason why they are named as they are).

File: make/Makefile.build.mk
Comment (jmazzitelli) [inline make/Makefile.build.mk:28]: What is the use-case for this target? How are people to use it? Note that all non-hidden targets (those not prefixed with a period `.`) have a `##` comment that is output when you do a `make help`... so add a `##` comment here and explain briefly this target and how it is to be used. I assume it should be something like this:

```suggestion
## Allows you to perform a multi-arch build. Set TARGET_ARCH env var to a value containing space-separated names of architectures you want built (e.g. TARGET_ARCH="amd64 arm64").
build-linux:
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Aside from my comments below, this failed on my box. I tried running  `make build container-build-kiali` and got this error:

```
$ make build container-build-kiali 
Building...
GOOS=linux GOARCH=amd64 CGO_ENABLED=0  go build \
	-o /home/jmazzite/source/kiali/kiali/bin/kiali -ldflags "-X main.version=v1.27.0-SNAPSHOT -X main.commitHash=fff76fb278ea2de0938052f3cba9717a6a5fb0f2"
Console version being packaged: 1.27.0-local-dc13a482979443fa9fa9807552f4a1c5de54c251
Preparing container image files
Building container image for Kiali using docker
docker build --pull --build-arg=TARGETARCH=amd64 -t quay.io/kiali/kiali:dev -f /home/jmazzite/source/kiali/kiali/src/github.com/kiali/kiali/_output/docker/Dockerfile /home/jmazzite/source/kiali/kiali/src/github.com/kiali/kiali/_output/docker
Sending build context to Docker daemon  56.66MB
Step 1/14 : FROM registry.access.redhat.com/ubi7-minimal AS base-amd64
latest: Pulling from ubi7-minimal
Digest: sha256:08f21393ea17f9122d245b77ffc03d99f6f25fe5d47b3ec246c31a76a318bf70
Status: Image is up to date for registry.access.redhat.com/ubi7-minimal:latest
 ---> 97012ae1b4ca
Step 2/14 : FROM registry.access.redhat.com/ubi8-minimal AS base-arm64
latest: Pulling from ubi8-minimal
Digest: sha256:c6473956faa4985a41776701993718eb27ab933078bf579768e8ca7d967459ee
Status: Image is up to date for registry.access.redhat.com/ubi8-minimal:latest
 ---> c103a05423dd
Step 3/14 : FROM registry.access.redhat.com/ubi8-minimal AS base-ppc64le
latest: Pulling from ubi8-minimal
Digest: sha256:c6473956faa4985a41776701993718eb27ab933078bf579768e8ca7d967459ee
Status: Image is up to date for registry.access.redhat.com/ubi8-minimal:latest
 ---> c103a05423dd
Step 4/14 : FROM base-${TARGETARCH:-amd64}
 ---> 97012ae1b4ca
Step 5/14 : LABEL maintainer="kiali-dev@googlegroups.com"
 ---> Using cache
 ---> c38448ec546b
Step 6/14 : ENV KIALI_HOME=/opt/kiali     PATH=$KIALI_HOME:$PATH
 ---> Using cache
 ---> 735266400160
Step 7/14 : WORKDIR $KIALI_HOME
 ---> Using cache
 ---> dfeeceffd8e7
Step 8/14 : RUN microdnf install -y shadow-utils &&     microdnf clean all &&     rm -rf /var/cache/yum &&     adduser --uid 1000 kiali
 ---> Using cache
 ---> 0d59164200ae
Step 9/14 : ARG TARGETARCH
 ---> Using cache
 ---> 8ffc46efa6c3
Step 10/14 : COPY kiali-${TARGETARCH} $KIALI_HOME/kiali
COPY failed: stat /var/lib/docker/tmp/docker-builder026032252/kiali-amd64: no such file or directory
make: *** [container-build-kiali] Error 1
```

**Comment by jmazzitelli** [comment]
note also (based on the output I'm seeing), even devs are going to have to pull down every arch image defined in the Dockerfile, which is not ideal (devs shouldn't have to be forced to download all the arch images).

File: deploy/docker/Dockerfile
Comment (morlay) [reply deploy/docker/Dockerfile:5]: TARGETARCH  is [platform arg](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope) and will automatic pass by docker. and this is important feature for `docker buildx build` to build mulit-arch images.

**Comment by morlay** [comment]
@jmazzitelli 

This may a docker issue. if care about downloading, we have to put  Dockerfile-ubi{7|8}-minimal back and add new file for multi-arch builds. but I think this may increase mantaince costs.

**Review by jmazzitelli — DISMISSED** [review]
I created a PR-to-this-PR here: https://github.com/morlay/kiali/pull/1
Changes I think need to be made before we can continue with a review.
I only tested DORP=docker (I do not have the "buildx" plugin or podman). To run with your buildx stuff, you set USE_DOCKER_BUILDX=true when running make.
See my additional comments in that PR-to-this-PR.

**Comment by morlay** [comment]
@jmazzitelli 

this PR is not for adding another tool for docker build. 
we use `buildx` to publish [multi-arch image](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/). example: https://quay.io/repository/morlay/kiali?tab=tags
image

i refactor and hope it could be clear.
```
$ IMAGE_ORG=morlay make build-linux-multi-arch container-multi-arch-push-kiali-quay
for arch in amd64 arm64 s390x ppc64le; do \
          GOOS=darwin GOARCH=amd64 CGO_ENABLED=0  go build \
           -o /Users/morlay/bin/kiali-${arch} -ldflags "-X main.version=v1.27.0-SNAPSHOT -X main.commitHash=a8a608ddeaee4812a2b6fb7bd8b62644b281d1ae"; \
        done
Console version being packaged: 1.26.0
Preparing container image files
Pushing Kiali multi-arch image to quay.io/morlay/kiali:dev using docker buildx
docker buildx build --push --platform=linux/amd64 --platform=linux/arm64 --platform=linux/s390x --platform=linux/ppc64le -t quay.io/morlay/kiali:dev -f /Users/morlay/src/github.com/kiali/kiali/_output/docker/Dockerfile-multi-arch /Users/morlay/src/github.com/kiali/kiali/_output/docker
```

**First step**: we need to compile all go binaries for all arch we want to supported

and copy the arch suffix binaries to `${OUTDIR}/docker`
```
_output/docker/
  console/
  Dockerfile-multi-arch
  kiali-amd64
  kiali-arm64
  kiali-ppc64le
  kiali-s390x
```

**Next step**: we use `buildx` for do build multi-arch image.

In this step we should do docker builds for all archs at same time. flags `--platform=linux/amd64 --platform=linux/arm64 --platform=linux/s390x --platform=linux/ppc64le` for this goal.

This part in Dockerfile, just make sure the multi-arch image not break RERL7 user as you said before
```
FROM registry.access.redhat.com/ubi7-minimal AS base-amd64
FROM registry.access.redhat.com/ubi8-minimal AS base-arm64
FROM registry.access.redhat.com/ubi8-minimal AS base-s390x
FROM registry.access.redhat.com/ubi8-minimal AS base-ppc64le

FROM base-${TARGETARCH:-amd64}
```

Then we could got the multi-arch image.

`docker pull` could automatic pull the host-arch-matched image. or use `docker pull --platform=linux/arm64` to pull the special arch image.

The final result is a new format with multi-arch info. 
```
$DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect quay.io/morlay/kiali:dev
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1789,
         "digest": "sha256:eb0dcf28e3d4029a69a0822da6233e535568f57f7a90fe8e130abcbd63de3471",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1789,
         "digest": "sha256:e468ba6db0be4bf484691e389fb0c3dbd73808b7862b9460ef73d9cbe915aca7",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1789,
         "digest": "sha256:51fd970cb8f87e2434e9618c225f39d5504fb7fac6240fc38b4d8ca0d1a3123d",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1789,
         "digest": "sha256:0202d48b27373f059d8fe9455a8f7962ca4a187485c7e56d58f571c003c2596f",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      }
   ]
}
```

File: make/Makefile.build.mk
Comment (jmazzitelli) [inline make/Makefile.build.mk:32]: Does this build the different arches? It looks like this just creates the same binary with different `-${arch}` in the names. GO_BUILD_ENVVARS will have GOARCH set to the same build host arch each time in the loop. Unless I misunderstand how this is working, I think you need to override GOARCH each time - something like this:
```suggestion
		${GO_BUILD_ENVVARS} GOARCH=${arch} ${GO} build \
			-o ${GOPATH}/bin/kiali-$${arch} -ldflags "-X main.version=${VERSION} -X main.commitHash=${COMMIT_HASH}"; \
```

Comment (morlay) [reply make/Makefile.build.mk]: Yes. i make a mistake here.
but without use $(MAKE). need to repeat GO_BUILD_ENVVARS `GOOS=linux GOARCH=$${arch} CGO_ENABLED=0`

updated.

File: make/Makefile.container.mk
Comment (morlay) [inline make/Makefile.container.mk:67]: could pass QUAY_TAG="quay.io/kiali/kiali:1.26 quay.io/kiali/kiali:1.26.1" in jenkins for multi tag publish

File: make/Makefile.container.mk
Comment (jmazzitelli) [inline make/Makefile.container.mk:66]: Because `buildx` is an experimental feature, we must ensure the user is OK that we are going to enable it. Also, `buildx` may not exist in the user's `docker` installation. Therefore, add a dependent target to ensure the user has `docker buildx` available and that the user has explicitly enabled it:
```suggestion
# Ensure "docker buildx" is available and enabled. For more details, see: https://github.com/docker/buildx/blob/master/README.md
.ensure-docker-buildx:
	@if docker buildx version > /dev/null 2>&1 || DOCKER_CLI_EXPERIMENTAL="${DOCKER_CLI_EXPERIMENTAL}" docker buildx version > /dev/null 2>&1 ; then \
	  echo "'docker buildx' is available and enabled."; \
	else \
	  if ! DOCKER_CLI_EXPERIMENTAL="enabled" docker buildx version > /dev/null 2>&1 ; then \
	    echo "You do not have 'docker buildx'. Note that 'buildx' is only available in Docker 19.03+"; \
	  else \
	    echo "You do not have 'docker buildx' enabled by default. Set DOCKER_CLI_EXPERIMENTAL=enabled if you want to use it."; \
	  fi; \
	  exit 1; \
	fi

## container-multi-arch-push-kiali-quay: Pushes the Kiali multi-arch image to quay.
container-multi-arch-push-kiali-quay: .ensure-docker-buildx .prepare-kiali-image-files
	@echo Pushing Kiali multi-arch image to ${QUAY_TAG} using docker buildx
	DOCKER_CLI_EXPERIMENTAL="enabled" docker buildx build --push $(foreach arch,${TARGET_ARCHS},--platform=linux/${arch}) $(foreach tag,${QUAY_TAG},--tag=${tag}) -f ${OUTDIR}/docker/Dockerfile-multi-arch ${OUTDIR}/docker
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Some additional things requested - see recommended suggestions in the comments.

File: make/Makefile.container.mk
Comment (morlay) [inline make/Makefile.container.mk:91]: Added .ensure-buildx-builder too.

For Mac user. The default builder couldn't using `--push`. need to create another builder. but multi arch supports provided by Docker for mac.

For Linux user.  need `qemu` to support multi-arch.

**Comment by morlay** [comment]
@jmazzitelli Thank your suggestions. I have learned a lot.

We could switch publishing job to use the simple command in CI (just make sure docker version 19.03+)
```
QUAY_TAG="quay.io/kiali/kiali:1.26 quay.io/kiali/kiali:1.26.2" make build-linux-multi-arch container-multi-arch-push-kiali-quay
```

File: make/Makefile.container.mk
Comment (morlay) [inline make/Makefile.container.mk:77]: `Docker@19.03.06` on Ubuntu 18.04 is not includes buildx too.

But could force install cli-plugin from binary-release
https://github.com/docker/buildx/blob/master/README.md#binary-release

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
My requested changes are described in this PR-to-this-PR: https://github.com/morlay/kiali/pull/2

**Review by cfcosta — COMMENTED** [review]
I'll give @jmazzitelli the last word on this, but on my machine:

I managed to build the images and they seem ok:

```
❯ file ../../../../bin/kiali*
../../../../bin/kiali:         ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=0gFu2K6jvnSBy2pG74p3/22Jycth7yKKWEjjjtPuj/jyWQwI45nhfIUNf_Ja9E/3RoKE14Y22EmDrCGaxio, not stripped
../../../../bin/kiali-amd64:   ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=0gFu2K6jvnSBy2pG74p3/22Jycth7yKKWEjjjtPuj/jyWQwI45nhfIUNf_Ja9E/3RoKE14Y22EmDrCGaxio, not stripped
../../../../bin/kiali-arm64:   ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=HruEc2Om6q3cigAxfH18/II4y9NcB1buwj9QApdc8/sw4mR1PMutA3W-7zy2xG/Ev8X34itw9ag55AVb53Z, not stripped
../../../../bin/kiali-ppc64le: ELF 64-bit LSB executable, 64-bit PowerPC or cisco 7500, version 1 (SYSV), statically linked, Go BuildID=8G5Ms24ZqTsNKUn-gdEC/Ob276cvtOSnxow6o_Cun/TLUdOA6b3ivBdwFXJiYJ/wV9NtaRSVOGkLTNQJZda, not stripped
../../../../bin/kiali-s390x:   ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=o3Y-aHsTiGjOsrP50Dcv/C2dWUVM6dk8RfsUxf10a/T0CKU4KRJMKdCzvTaMce/BGn5Eooox7YgBXa51BBp, not stripped
```
I have not tested pushing the images to quay.io so can not comment on that. Code seems ok, on my machine I didn't need to install docker-buildx (arch, x86_64).

**Comment by morlay** [comment]
Just notice, may have some permission issue for quay.io

I got 401 error when I push images to quay.io, no idea why. with an Admin robot
```
QUAY_TAG="quay.io/morlay/kiali:1.26 quay.io/morlay/kiali:1.26.1" make build-linux-multi-arch container-multi-arch-push-kiali-quay
```

But works well to push docker.io
```
QUAY_TAG="morlay/kiali:1.26 morlay/kiali:1.26.1" make build-linux-multi-arch container-multi-arch-push-kiali-quay
```

**Comment by jmazzitelli** [comment]
My machine is unable to build anything other than amd64 for some reason, but I was able to successfully build and push to quay.io using the following command:
```
 TARGET_ARCHS="amd64" DOCKER_CLI_EXPERIMENTAL=enabled IMAGE_ORG=my-quay-username make build-linux-multi-arch container-multi-arch-push-kiali-quay
```

**Comment by jmazzitelli** [comment]
Before merging, can anyone else on the team confirm `container-multi-arch-push-kiali-quay` make target is successful for them when truly building a multi-arch image (I can only get it to work with amd64, but I think that may just be a limitation on my laptop).

**Comment by jshaughn** [comment]
I tried running this and hit:

```
DOCKER_CLI_EXPERIMENTAL="enabled" docker buildx build --push --builder=kiali-builder --platform=linux/amd64 --platform=linux/arm64 --platform=linux/s390x --platform=linux/ppc64le --tag=quay.io/jshaughn/kiali:dev -f /home/jshaughn/go-kiali/src/github.com/kiali/kiali/_output/docker/Dockerfile-multi-arch /home/jshaughn/go-kiali/src/github.com/kiali/kiali/_output/docker
unknown flag: --builder
```

I'm on F29 and my docker version is 19.03.5

**Comment by jmazzitelli** [comment]
> I'm on F29 and my docker version is 19.03.5

Looks like we need something more than just "19.03" - I'm at "19.03.13" and I have that option. I'll submit a code change to do a check and make sure you have a valid docker version and if not the build should abort.

File: make/Makefile.container.mk
Comment (jmazzitelli) [inline make/Makefile.container.mk:67]:
```suggestion
.ensure-docker-buildx:
	@required_docker="19.03.13"; \
	current_docker="$$(docker version --format '{{.Client.Version}}')"; \
	if [ "$$(printf $$required_docker\\n$$current_docker | sort -V | head -n1)" != "$$required_docker" ]; then \
	  echo "You have an older 'docker' [$$current_docker] which is not able to perform this build. Please upgrade to at least [$$required_docker]"; \
	  exit 1; \
	else \
	  echo "Docker is version [$$current_docker] which is the same or newer than the required [$$required_docker] - build can continue"; \
	fi
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Make sure the user has a docker version that is compatible. If the user does not, abort the build.

File: make/Makefile.container.mk
Comment (morlay) [reply make/Makefile.container.mk:67]: @jshaughn 's issue not for docker. I'm not sure which OS he used. may MacOS ?

docker desktop includes a buildx. but not includes in docker for Linux distro

```
$ docker buildx version
github.com/docker/buildx v0.4.2 fb7b670b764764dc4716df3eba07ffdae4cc47b2
```

so here just need to check the buildx version

Comment (jmazzitelli) [reply make/Makefile.container.mk:67]: Ahh.. ok, so we should be looking at the buildx version, not the docker version. That makes sense because his error was his buildx did not accept the `--builder` option. So we just have to tweek my suggested fix to look at the buildx version and not the docker version.

Comment (morlay) [reply make/Makefile.container.mk:67]: Yeah. I'm trying to do this.

File: make/Makefile.container.mk
Comment (morlay) [inline make/Makefile.container.mk:79]: import use this. we need os and arch match for host not for makefile
```
$$(go env GOOS)-$$(go env GOARCH)
```

use this to trigger install for local test
```
mkdir -p ${HOME}/.docker/cli-plugins && touch ${HOME}/.docker/cli-plugins/docker-buildx
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
After seeing the error @jshaughn got and the latest changes, I have another round of suggested changes. See https://github.com/morlay/kiali/pull/3

**Review by jmazzitelli — APPROVED** [review]
This now looks good.

**Comment by jmazzitelli** [comment]
I'm trying to kick-start another travis build for this PR; something in travis seems hung or broken.

**Comment by jmazzitelli** [comment]
Something is weird with the travis integration - I'm going to close this PR and then immediately re-open it to see if that wakes things up.

**Comment by morlay** [comment]
@jmazzitelli 

is the final images publishing in jenkins-ci ? 
could you take over to update it ? I'm not be familiar with jenkins-ci.

**Comment by jmazzitelli** [comment]
We did not do any changes to our current jenkins jobs. At the moment, these targets are unused by the Kiali team.

**Comment by morlay** [comment]
so still not publish multi arch image on next release （1.28.0） ？

**Comment by jmazzitelli** [comment]
> so still not publish multi arch image on next release （1.28.0） ？

We do not yet have multi-arch builds for the operator. I do not think we want to publish multi-arch images for the server without also having the same multi-arch support with the operator. So we will want to have a multi-arch operator build to go along with this multi-arch server build - otherwise, folks running Kiali in production with the operator (which is the recommended deployment model) will be confused why they can't do so on non-amd64 hardware. I created a github issue for this task here: https://github.com/kiali/kiali/issues/3460

Also, see @israel-hdez [comment in the original issue](https://github.com/kiali/kiali/issues/1091#issuecomment-630238473) regarding our need to do some research to find out if experimental buildx is even available on our jenkins server at this time.

Finally, as a side note, the Kiali team is currently not prepared to officially support any arch's other than amd64 at the moment. Before we do any publishing of multi-arch images to our public quay repository, we are going to have to publish some kind of statement saying these non-amd64 images are not officially supported (see https://github.com/kiali/kiali/issues/1091#issuecomment-630208763)

**Comment by morlay** [comment]
upstream  `quay.io/operator-framework/ansible-operator:v0.17.0` is already multi arch image.

no need to change anything. 
only use buildx to build multi arch image with Dockerfile of kiali-operator

**Comment by jmazzitelli** [comment]
> upstream  `quay.io/operator-framework/ansible-operator:v0.17.0` is already multi arch image.
> 
> no need to change anything.
> only use buildx to build multi arch image with Dockerfile of kiali-operator

Put that comment in https://github.com/kiali/kiali/issues/3460 - that is where we are tracking the task to build operator multi-arch image.

**Comment by jmazzitelli** [comment]
followup issue: https://github.com/kiali/kiali/issues/3463

**Pull request merged**
